File: C++/Medium/WrongAnswer/most-popular-video-creator_0.cpp
class Solution {
public:
    vector<vector<string>> mostPopularCreator(vector<string>& creators, vector<string>& ids, vector<int>& views) {
        
        unordered_map<string, pair<int, string>> m;
        for (int i = 0; i < creators.size(); ++i) {
            if (m.find(creators[i]) == m.end() || m[creators[i]].first < views[i]) {
                m[creators[i]] = {views[i], ids[i]};
            } else if (m[creators[i]].first == views[i] && m[creators[i]].second > ids[i]) {
                m[creators[i]] = {views[i], ids[i]};
            }
        }
        int maxView = 0;
        for (auto& it : m) {
            maxView = max(maxView, it.second.first);
        }
        vector<vector<string>> res;
        for (auto& it : m) {
            if (it.second.first == maxView) {
                res.push_back({it.first, it.second.second});
            }
        }
        return res;
    }
};

File: Python3/Medium/WrongAnswer/most-popular-video-creator_0.py
class Solution:
    def mostPopularCreator(self, creators: List[str], ids: List[str], views: List[int]) -> List[List[str]]:
        
        d = {}
        for i in range(len(creators)):
            if creators[i] not in d:
                d[creators[i]] = [(ids[i], views[i])]
            else:
                d[creators[i]].append((ids[i], views[i]))
        max_views = 0
        res = []
        for key in d:
            views = 0
            max_id = ""
            for id, view in d[key]:
                if view > views:
                    views = view
                    max_id = id
                elif view == views:
                    if id < max_id:
                        max_id = id
            if views > max_views:
                max_views = views
                res = [[key, max_id]]
            elif views == max_views:
                res.append([key, max_id])
        return res